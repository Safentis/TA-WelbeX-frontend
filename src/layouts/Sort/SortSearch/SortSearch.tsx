import React, { FC } from 'react';

import './SortSearch.css';
import { Props } from './SortSearch.interface';

/**
 * SortSearch component
 * @returns {React.ReactElement} - returns react element
 */

const SortSearch: FC<Props> = ({}): React.ReactElement => {
  return (
    <ul className="sort-search sort__nested">
      <li className="sort-search__item">
        <input type="text" />
      </li>
    </ul>
  );
};

export default SortSearch;
