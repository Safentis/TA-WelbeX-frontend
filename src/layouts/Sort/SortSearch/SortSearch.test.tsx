import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import SortSearch from './SortSearch';

import { Props } from './SortSearch.interface';

const setUp = (props: Props) => {
  return shallow(<SortSearch {...props} />);
};

describe('<SortSearch />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;

  beforeEach(() => {
    component = setUp({});
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 ul element', () => {
      expect(component.find('ul')).toBeTruthy();
    });

    it('has 1 li element', () => {
      expect(component.find('li')).toBeTruthy();
    });

    it('has 1 input element', () => {
      expect(component.find('input')).toBeTruthy();
    });
  });
});
