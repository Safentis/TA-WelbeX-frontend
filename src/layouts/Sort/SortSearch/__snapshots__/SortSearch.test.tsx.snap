// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortSearch /> component does match to the snapshot 1`] = `
<ul
  className="sort-search sort__nested"
>
  <li
    className="sort-search__item"
  >
    <input
      type="text"
    />
  </li>
</ul>
`;
